internal/utils_test.go

```diff
@@ -266,3 +279,97 @@ func (token *mockedToken) Done() <-chan struct{} {
 func (token *mockedToken) Error() error {
 	return token.err
 }
+
+func createLocalArtifact(t *testing.T, dir, alias, body string) (file string, hash string) {
+	t.Helper()
+
+	file = filepath.Join(dir, alias)
+	if err := storage.WriteLn(file, body); err != nil {
+		t.Fatal("error writing to file", err)
+	}
+	hType := sha256.New()
+	hType.Write([]byte(body))
+	hash = hex.EncodeToString(hType.Sum(nil))
+	return
+}
+
+func convertLocalArtifact(path, name, hash string, len int) *hawkbit.SoftwareArtifactAction {
+	return &hawkbit.SoftwareArtifactAction{
+		Filename: name,
+		Download: map[hawkbit.Protocol]*hawkbit.Links{
+			storage.ProtocolFile: {URL: path},
+		},
+		Checksums: map[hawkbit.Hash]string{
+			hawkbit.SHA256: hash,
+		},
+		Size: len,
+	}
+}
+
+func getAbsolutePath(t *testing.T, path string) string {
+	t.Helper()
+
+	absolutePath, err := filepath.Abs(path)
+	if err != nil {
+		t.Fatalf("error getting absolute file path of %v - %v", path, err)
+	}
+	return absolutePath
+}
+
+func checkFileExistsWithContent(t *testing.T, filename, expectedContent string) {
+	content, err := os.ReadFile(filename)
+	if err != nil {
+		t.Fatalf("expected file %s to be created from install script(error opening: %v)", filename, err)
+	}
+	message := strings.Trim(string(content), "\r\n")
+	if expectedContent != message {
+		t.Fatalf("wrong install script execution result, expected: %s, got: %s", expectedContent, message)
+	}
+}
+
+func checkNoFilesCopied(t *testing.T, dir string, canBeMoved bool) bool {
```

Comment:
canBeMoved seems be be very tricky. I would suggest to integrate more stable approach.